### PR TITLE
Additional label for ratelimit error cause

### PR DESCRIPTION
**Pull request by noerw**

### CODE REVIEW

File: internal/client/client.go [hunk 1]
```diff
@@ -50,6 +51,9 @@ func New(infoURL, username, password, authToken string, timeout time.Duration, u
 		if res.StatusCode == http.StatusUnauthorized {
 			return nil, ErrNotAuthorized
 		}
+		if res.StatusCode == http.StatusTooManyRequests {
```
Comment (xperimental) [inline internal/client/client.go:54]: I think four cases is the point where I would replace the if-blocks with a switch.

**Review by xperimental — CHANGES_REQUESTED** [review]
Thanks for the contribution, this seems like a useful idea.

I have one small change idea.

**Review by xperimental — CHANGES_REQUESTED** [review]
I made a few other changes to the code yesterday. Can you rebase and add a test-case?

**Comment by nivadis** [comment]
how does this come up? because i'm also facing this 429 errors, with 15s scrape, understandable. but I haven't found changes causing this now.

sorry for this little bit OT

**Comment by xperimental** [comment]
@noerw It has been a while, but I have not checked in for some time as well...

Do you want to continue on this PR?

**Comment by noerw** [comment]
I'm sure I rebased this change, but looks like I didn't push it. I'll push this once I'm home again in 2 weeks

**Review by xperimental — APPROVED** [review]

**Pull request merged**